src/components/include/protocol_handler/session_observer.h

```diff
@@ -64,21 +67,21 @@ class SessionObserver {
   /**
    * \brief Callback function used by ProtocolHandler
    * when Mobile Application initiates start of new session.
+   * Result must be notified through NotifySessionStartedResult().
    * \param connection_handle Connection identifier within which session
    * has to be started.
    * \param sessionId Identifier of the session to be start
    * \param service_type Type of service
    * \param protocol_version Version of protocol
    * \param is_protected would be service protected
-   * \param hash_id pointer for session hash identifier, uint32_t* hash_id
-   * \return uint32_t Id (number) of new session if successful, otherwise 0.
+   * \param params configuration parameters specified by mobile
    */
-  virtual uint32_t OnSessionStartedCallback(
+  virtual void OnSessionStartedCallback(
```

Comment:
Done in https://github.com/smartdevicelink/sdl_core/pull/1700/commits/9f0795170b9cc46c6c8565ba967da6ae3bdbc613.